internal/discover/hooks.go

```diff
@@ -44,26 +44,66 @@ func (h *Hook) Hooks() ([]Hook, error) {
 	return []Hook{*h}, nil
 }
 
+type HookName string
+
+// DisabledHooks allows individual hooks to be disabled.
+type DisabledHooks map[HookName]bool
+
+func (d DisabledHooks) Set(value HookName) {
+	if value == "all" {
+		for _, hook := range AllHooks {
+			d[hook] = true
+		}
+		return
+	}
+
+	d[value] = true
+}
```

Comment:
Instead of this, we can add an `isDisabled(HookName)` function (something like we had before in `IsHookSupported` that has been removed). We can impliment it as follows:
```
// HookIsSupported checks whether a hook of the specified name is supported.
// Hooks must be explicitly disabled, meaning that if no disabled hooks are
// all hooks are supported.
func (d disabledHooks) isDisabled(h HookName) bool {
	if d["all"] {
		return true
	}
	return d[h]
} 
```

Reply:
func removed, it was not needed by moving to a options implementation